config/example-menu.cfg

```diff
@@ -3,153 +3,116 @@
 # The snippets in this file may be copied into the main printer.cfg file.
 # See the "example.cfg" file for description of common config parameters.
 
-# Available menu elements:
-#   item - purely visual element
-#   command - same like 'item' but with gcode trigger
-#   input - same like 'command' but has value changing capabilities
-#   list - menu element container, with entry and exit gcode triggers
-#   vsdcard - same as 'list' but will append files from virtual sdcard
+# Menu has default variables defined in menu.cfg
+# For example, accessible as printer.menu.default.hotend_target_max
+# Variables can be modified in realtime by using 'Setup -> Edit defaults' menu
+# Following template actions can be used:
+#   printer.menu.action_set_default(name, value) - will set <value> for default variable <name>
+#   printer.menu.action_reset_defaults() - will reset saved user default values to the factory defaults
+# Default variables are accessible from the script as {printer.menu.default.default_variable_name}
+# Don't forget to save your config by using 'Setup -> Save config' after modifing defaults.
+#
+# Menu also has an option for temporary runtime variables.
+# Following template action can be used:
+#   printer.menu.action_set_runtime(name, value) - will set <value> for runtime variable <name>
+# Runtime variables are accessible from the script as {printer.menu.runtime.runtime_variable_name}
+# The runtime variables are not stored between restarts.
```

Comment:
I don't think we should add any menu actions or allow any menu specific runtime configuration.  If a user needs to add something with state then they can add custom menu items that refer to their gcode_macro variables.